### PR TITLE
Refactor time utils and create a separate package

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
### Checklist

- [x] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

**Comment by pavelkalinnikov** [comment]
PTAL. Best to review commit by commit.

**Comment by Martin2112** [comment]
This might need a changelog update.

File: util/clock/timesource.go [hunk 1]
```diff
@@ -106,18 +110,19 @@ func (f *FakeTimeSource) Set(t time.Time) {
...
-// IncrementingFakeTimeSource takes a base time and several increments, which will be applied to
-// the base time each time Now() is called. The first call will return the base time + zeroth
-// increment. If called more times than provided for then it will panic. Does not require that
-// increments increase monotonically.
+// IncrementingFakeTimeSource takes a base time and several increments, which
+// will be applied to the base time each time Now() is called. The first call
+// will return the base time + zeroth increment. If called more times than
+// provided for then it will panic. Does not require that increments increase
+// monotonically.
 type IncrementingFakeTimeSource struct {
```
Comment (Martin2112) [inline util/clock/timesource.go]: Can probably shorten this name as well now. Can be done later.

Comment (pavelkalinnikov) [reply util/clock/timesource.go]: Done. Please review the last commit.

**Comment by Martin2112** [comment]
Did you fix the metalint errors? See travis build.

**Comment by pavelkalinnikov** [comment]
Not sure I understand the cause of these errors. The file is gofmt-ed on my side.

**Comment by Martin2112** [comment]
Different version of the linter possibly?

**Comment by Martin2112** [comment]
Or there might be some existing problem in the file that was added before we had the linter set up.

**Comment by Martin2112** [comment]
Hmmmm. Not seeing anything when I run gofmt on it.

**Comment by pavelkalinnikov** [comment]
#1381

**Comment by pavelkalinnikov** [comment]
For this PR, I restored the original formatting. We can deal with formatting problem separately in #1381, or procrastinate until it happens again :)

**Pull request merged**